src/cmd/fetch.go

```diff
@@ -20,29 +21,72 @@ var fetchCmd = &cobra.Command{
 	2. A branch name
 	3. A path to a whitelist file
 	`,
-	
-	Args: cobra.MinimumNArgs(3),
+
+	//Args: cobra.MinimumNArgs(3),
 
 	Run: func(cmd *cobra.Command, args []string) {
-		if len(args) % 3 != 0 {
+		branch, err := rootCmd.Flags().GetString("branch")
+		if err != nil {
+			panic(err)
+		}
+		whitelist, err := rootCmd.Flags().GetString("whitelist")
+		if err != nil {
+			panic(err)
+		}
+
+		if len(args)%3 != 0 && branch == "" && whitelist == "" {
 			panic("arguments is not divisable by 3")
-		} 
-		Fetcher.ListingReposForFetch(args)
-		
+		}
+
+		if branch != "" && whitelist != "" {
+			// construct the args array with branch and whitelist
+			temp_args := []string{}
+			for i := 0; i < len(args); i++ {
+				temp_args = append(temp_args, args[i])
+				temp_args = append(temp_args, branch)
+				temp_args = append(temp_args, whitelist)
+			}
+			args = temp_args
+		} else {
+
+			// construct the args array based on individual flags
+			temp_args := []string{}
+			if branch != "" {
+				for i := 0; i < len(args); i++ {
+					// inserts branch at every 2nd index and shifts the rest of the values
+					if (i+1)%2 == 0 {
+						temp_args = append(temp_args, branch)
+						temp_args = append(temp_args, args[i])
+
+					} else {
+						temp_args = append(temp_args, args[i])
+					}
+				}
+				args = temp_args
+			}
+			if whitelist != "" {
+				for i := 0; i < len(args); i++ {
+					if (i+1)%2 == 0 {
+						temp_args = append(temp_args, args[i])
+						temp_args = append(temp_args, whitelist)
+					} else {
+						temp_args = append(temp_args, args[i])
+					}
+				}
+				args = temp_args
+			}
+		}
+		err = Fetcher.ListingReposForFetch(args)
+		if err != nil {
+			log.Print(err)
+		}
 		fmt.Println("fetch called")
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(fetchCmd)
 
-	// Here you will define your flags and configuration settings.
-
-	// Cobra supports Persistent Flags which will work for this command
-	// and all subcommands, e.g.:
-	// fetchCmd.PersistentFlags().String("foo", "", "A help for foo")
-
-	// Cobra supports local flags which will only run when this command
-	// is called directly, e.g.:
-	// fetchCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	rootCmd.PersistentFlags().String("branch", "", "Branch name for all repositories")
+	rootCmd.PersistentFlags().String("whitelist", "", "Path to a whitelist file for all repositories")
```

Comment:
Is there a reason that these are `PersistenFlags`?

Reply:
I tried with normal flags but for some reason it didn't work when i tried to parse them. If we can figure out a fix then that would be great

Reply:
Might it have smth to do with inheritance? Ex:

> Local flags aren't inherited by subcommands or available to commands at the same level. On the other hand, you can define persistent flags at the root level of a command for subcommands to inherit. Then, subcommands of the root command can access the persistent flag without needing redefinition.
\- [How  To Use the Cobra Package in Go](https://www.digitalocean.com/community/tutorials/how-to-use-the-cobra-package-in-go)

Reply:
That could be the case def had some problems with the flags but I defined it inside the command file so no clue why it didn't work since it's not a subcommand at that point :shrug: This works tho so I guess we can leave it for now and figure it out later if somebody has the time and know how

Reply:
I have fixed this. See my [commit](https://github.com/Agile-Arch-Angels/decentralized-tech-radar_dev/pull/28/commits/a546d3bf4e165bcebfef93f711f80056a4e72a67)